src/Groups/directproducts.jl

```diff
@@ -213,21 +193,21 @@ function _as_subgroup(G::DirectProductGroup, H::GapObj)
   return _as_subgroup(G, H, GAPGroupElem{DirectProductGroup})
 end
 
-function sub(G::DirectProductGroup, elms::Vector{GAPGroupElem{DirectProductGroup}})
+function sub(G::DirectProductGroup, elms::Vector{<:GAPGroupElem{DirectProductGroup}})
   elems_in_GAP = GAP.julia_to_gap(GapObj[x.X for x in elms])
-  H = GAP.Globals.Group(elems_in_GAP)
+  H = GAP.Globals.Subgroup(G.X,elems_in_GAP)
   #H is the group. I need to return the inclusion map too
   return _as_subgroup(G, H)
 end
```

Comment:
I think it would be simpler to define
```
function sub(G::T, elms::Vector{<:GAPGroupElem{T}}) where T<:GAPGroup
  elems_in_GAP = GAP.julia_to_gap(GapObj[x.X for x in elms])
  H = GAP.Globals.Subgroup(G.X,elems_in_GAP)
  #H is the group. I need to return the inclusion map too
  return _as_subgroup(G, H)
end
```
This would cover also the `sub` methods for semidirect products and wreath products.

Reply:
Indeed, I have just checked that for products of groups (direct, semidirect, wreath), I need to redefine only the method `_as_subgroup_bare` (because the constructors are different). The rest is not necessary and I can remove it.